components/support/error/src/macros.rs

```diff
@@ -60,3 +60,19 @@ macro_rules! breadcrumb {
         }
     };
 }
+
+/// Function wrapper macro to convert from a component's internal errors to external errors
+/// and optionally log and report the error.
+#[macro_export]
+macro_rules! handle_error {
+    { $($tt:tt)* } => {
+        let body = || {
+            $($tt)*
+        };
+        let result: Result<_> = body();
+        match result {
+            Ok(r) => Ok(r),
+            Err(e) => Err($crate::convert_log_report_error(e))
+        }
```

Comment:
nit: we can use `map_err` here
```suggestion
        result.map_err($crate::convert_log_report_error)
```